code/game/objects/structures/guncase.dm

```diff
@@ -60,36 +60,58 @@
 	if(iscyborg(user) || isalien(user))
 		return
 	if(contents.len && open)
-		ShowWindow(user)
+		show_menu(user)
 	else
 		open = !open
 		update_icon()
 
-/obj/structure/guncase/proc/ShowWindow(mob/user)
-	var/dat = {"<div class='block'>
-				<h3>Stored Guns</h3>
-				<table align='center'>"}
-	if(LAZYLEN(contents))
-		for(var/i in 1 to contents.len)
-			var/obj/item/I = contents[i]
-			dat += "<tr><A href='?src=[REF(src)];retrieve=[REF(I)]'>[I.name]</A><br>"
-	dat += "</table></div>"
+/**
+  * show_menu: Shows a radial menu to a user consisting of an available weaponry for taking
+  *
+  * Arguments:
+  * * user The mob to which we are showing the radial menu
+  */
+/obj/structure/guncase/proc/show_menu(mob/user)
+	if(!LAZYLEN(contents))
+		return
 
-	var/datum/browser/popup = new(user, "gunlocker", "<div align='center'>[name]</div>", 350, 300)
-	popup.set_content(dat)
-	popup.open(FALSE)
+	var/list/display_names = list()
+	var/list/items = list()
+	for(var/i in 1 to contents.len)
+		var/obj/item/I = contents[i]
+		display_names[I.name + " ([i])"] = REF(I)
+		var/image/item_image = image(icon = I.icon, icon_state = I.icon_state)
+		if(length(I.overlays))
+			item_image.copy_overlays(I)
+		items += list(I.name + " ([i])" = item_image)
 
-/obj/structure/guncase/Topic(href, href_list)
-	if(href_list["retrieve"])
-		var/obj/item/O = locate(href_list["retrieve"]) in contents
-		if(!O || !istype(O))
-			return
-		if(!usr.canUseTopic(src, BE_CLOSE) || !open)
-			return
-		if(ishuman(usr))
-			if(!usr.put_in_hands(O))
-				O.forceMove(get_turf(src))
-			update_icon()
+	var/pick = show_radial_menu(user, src, items, custom_check = CALLBACK(src, .proc/check_menu, user), radius = 36, require_near = TRUE)
+	if(!pick)
+		return
+
+	var/weapon_reference = display_names[pick]
+	var/obj/item/O = locate(weapon_reference) in contents
+	if(!O || !istype(O))
+		return
+	if(ishuman(user))
```

Comment:
early return

Reply:
I put the check in the check_menu proc instead, with an early return. Fixed.